src/compiler/checker.ts

```diff
@@ -2014,6 +2017,34 @@ module ts {
             }
         }
 
+        // Push an entry on the type resolution stack. If an entry with the given target is not already on the stack,
+        // a new entry with that target and an associated result value of true is pushed on the stack, and the value
+        // true is returned. Otherwise, a circularity has occurred and the result values of the existing entry and
+        // all entries pushed after it are changed to false, and the value false is returned. The target object has
+        // no significance other than to provide a unique identity for a particular type resolution result.
+        function pushTypeResolution(target: Object): boolean {
+            for (let i = 0; i < resolutionCount; i++) {
+                if (resolutionTargets[i] === target) {
+                    do {
+                        resolutionResults[i++] = false;
+                    }
+                    while (i < resolutionCount);
+                    return false;
+                }
+            }
+            resolutionTargets[resolutionCount] = target;
+            resolutionResults[resolutionCount] = true;
+            resolutionCount++;

```

Comment:
i'd say yes, because during the 'pop' operation we don't resize these arrays but rather replace the value with `undefined` so `length` property does not reflect the actual amount of data that is stored in arrays